Mathlib/Data/Set/Intervals/Basic.lean

```diff
@@ -1947,3 +1947,38 @@ instance : NoMaxOrder (Set.Iio x) :=
     exact ⟨⟨b, hb₂⟩, hb₁⟩⟩
 
 end Dense
+
+/-! ### Lemmas about intervals in subtypes
+-/
+
+theorem Set.image_subtype_val_Icc_subset [Preorder α] {p : α → Prop} (a b : {x // p x}) :
```

Comment:
I would use `MapsTo Subtype.val (Icc a b) (Icc a.val b.val)`, then `(mapsTo_subtype_val_Icc _ _).image_subset` whenever you need the original statement. BTW, do we have these lemmas for any `Monotone`/`StrictMono` map?

Reply:
I've added all the `mapsTo` lemmas requested, as well as `Monotone.image_Icc_subset` and friends, but I've kept the original lemmas too.